docs/src/pages/guides/FHIRServerUsersGuide.md

```diff
@@ -1098,6 +1099,135 @@ team can more easilly read the messages.
+### 4.10.1 BulkData Export
+BulkData export is implemented according to [Fhir r4 BulkData spec](http://hl7.org/fhir/uv/bulkdata/STU1/export/index.html).   
+There are 2 projects involved inside the implementation:
+- fhir-operation-bulkdata
+- fhir-bulkimportexport-webapp   
+
+(There is a ExportOperationTest.java in fhir-server-test project with server integration test cases for system, patient and group export)  
+The fhir-operation-bulkdata project implements the REST APIs for Bulkdata export as FHIR operations, there are ExportOperation, PatientExportOperation and GroupExportOperation defined for system export, patient export and group export accordingly. And these operations call JavaBatch jobs defined in the fhir-bulkimportexport-webapp project to do the real export work.   
+There are 3 chunk style JavaBatch jobs defined as following in fhir-bulkimportexport-webapp project for the above 3 export operations:  
+- FhirBulkExportChunkJob
+- FhirBulkExportPatientChunkJob
+- FhirBulkExportGroupChunkJob
+
+The fhir-bulkimportexport-webapp project is also a wrapper for the whole BulkData web application, which generates fhir-bulkimportexport.war, this war should be copied to the apps directory of the liberty fhir-server instance. Following is a sample deployment description for fhir-bulkimportexport.war in server.xml of liberty server:
+
+```
+    <webApplication id="fhir-bulkimportexport-webapp" location="fhir-bulkimportexport.war" name="fhir-bulkimportexport-webapp">
+        <classloader commonLibraryRef="fhirSharedLib" privateLibraryRef="configResources,fhirUserLib"/>
+        <application-bnd>
+            <security-role id="users" name="FHIRUsers">
+                <group name="FHIRUsers"/>
+            </security-role>
+        </application-bnd>
+    </webApplication>
+```
+
+The exported FHIR resources can be configured to be written into IBM COS or Amazon S3 bucket in the per-tenant bulkdata.json configuration file which is put under the tenant configuration directory of the fhir-server instance. Following is a sample bulkdata.json which is configured to export the FHIR resources into fhir-bulkdata-sample bucket of IBM COS: 
+
+```
+{
+  "applicationName": "fhir-bulkimportexport-webapp",
+  "moduleName": "fhir-bulkimportexport.war",
+  "jobParameters": {
+    "cos.bucket.name": "fhir-bulkdata-sample",
+    "cos.location": "us",
+    "cos.endpointurl": "https://s3.us-south.cloud-object-storage.appdomain.cloud",
+    "cos.credential.ibm": "Y",
+    "cos.api.key": "xxxxxxxxxxxxxxxxxxxxxxx",
+    "cos.srvinst.id": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+  },
+  "implementation_type": "cos",
+  "batch-uri": "https://localhost:9443/ibm/api/batch/jobinstances",
+  "batch-user": "fhiradmin",
+  "batch-user-password": "xxxxxxxxxxxx",
+  "batch-truststore" : "resources/security/fhirTruststore.jks",
+  "batch-truststore-password" : "xxxxxxxxxxxx",
+  "serverHostname" : "localhost:9443",
+    "contextRoot" : "/fhir-server/api/v4"
+}
```

Comment:
maybe the values that can be set above should be described for the client? 
Also cos.bucket.prefix ? it seems pretty useful.

Reply:
good point, will add. cos.bucket.prefix is generated automatically by codes, so I didn't add it.